internal/command/unlock.go

```diff
@@ -95,6 +100,17 @@ func (c *UnlockCommand) Run(args []string) int {
 
 	_, isLocal := stateMgr.(*statemgr.Filesystem)
 
+	if backendConfig != nil && (backendConfig.Type == "s3" || backendConfig.Type == "oss" || backendConfig.Type == "http") {
```

Comment:
I think we don't need to check configuration values here. Let's rely on the interface assertion check. Interface assertion + `IsLockingEnabled` call will cover it.